src/core/lpLexer.ml

```diff
@@ -0,0 +1,279 @@
+
+  val unquote : string -> string
+  (** [unquote s] removes the quotation marks [{|] and [|}] from [s] if it has
+      some. Otherwise, the argument is returned. *)
+
+  val is_debug_flag : string -> bool
+  (** [is_debug_flag s] is true if [s] is a debug flag. *)
+
+  val lexer : lexbuf -> unit -> token * Lexing.position * Lexing.position
+  (** [lexer buf] is a lexing function on buffer [buf] that can be passed to
+      a parser. *)
+end = struct
+  let digit = [%sedlex.regexp? '0' .. '9']
+  let integer = [%sedlex.regexp? Plus digit]
+
+  (* We define the set of UTF8 codepoints that make up identifiers. The
+     builtin categories are described on the home page of sedlex
+     @see https://github.com/ocaml-community/sedlex *)
+
+  let superscript =
+    [%sedlex.regexp? 0x2070 | 0x00b9 | 0x00b2 | 0x00b3 | 0x2074 .. 0x207c]
+  let subscript = [%sedlex.regexp? 0x208 .. 0x208c]
+  let supplemental_punctuation = [%sedlex.regexp? 0x2e00 .. 0x2e52]
+  let alphabet = [%sedlex.regexp? 'a' .. 'z' | 'A' .. 'Z']
+  let ascii_sub =
+    [%sedlex.regexp? '-' | '\'' | '&' | '^' | '\\' | '*'
+                   | '%' | '#' | '~']
+  let letter =
+    [%sedlex.regexp? lowercase | uppercase | ascii_sub
+                   | math | other_math | subscript | superscript
+                   | supplemental_punctuation ]
+  let id = [%sedlex.regexp? (letter | '_'), Star (letter | digit | '_')]
+  let escid =
+    [%sedlex.regexp? "{|", Star (Compl '|' | '|', Compl '}'), Star '|', "|}"]
+  let stringlit = [%sedlex.regexp? '"', Star (Compl ('"' | '\n')), '"']
+  let float = [%sedlex.regexp? integer, '.', Opt (integer)]
+  let comment = [%sedlex.regexp? "//", Star (Compl ('\n' | '\r'))]
+
+  (** [nom buf] eats whitespaces and comments in buffer [buf]. *)
+  let rec nom : lexbuf -> unit = fun buf ->
+    match%sedlex buf with
+    | ' ' -> nom buf
+    | '\t' -> nom buf
+    | '\n' -> nom buf
+    | '\r' -> nom buf
+    | "\r\n" -> nom buf
+    | "/*" -> nom_comment buf
+    | comment -> nom buf
+    | _ -> ()
+and nom_comment : lexbuf -> unit = fun buf ->
+    match%sedlex buf with
+    | eof -> raise (SyntaxError (Pos.none "Unterminated comment."))
+    | "*/" -> nom buf
+    | any -> nom_comment buf
+    | _ -> assert false
+
+  (** [is_escaped s] is true if string [s] is escaped: i.e. enclosed in "{|
+      |}". *)
+  let is_escaped : string -> bool = fun s ->
+    let buf = Utf8.from_string s in
+    match%sedlex buf with
+    | escid -> true
+    | _ -> false
+
+  let unquote : string -> string = fun s ->
+    if is_escaped s then String.(sub s 2 (length s - 4)) else s
+
+  (** [is_identifier s] is true if [s] is an identifier. *)
+  let is_identifier : string -> bool = fun s ->
+    let lexbuf = Sedlexing.Utf8.from_string s in
+    match%sedlex lexbuf with
+    | id -> true
+    | _ -> false
+
+  let is_keyword : string -> bool = fun s ->
+    (* NOTE this function may be optimised using a map, a hashtable, or using
+       [match%sedlex]. *)
```

Comment:
Do you plan to do it in this PR?

Reply:
I don't think so, to do it properly, we'd have to time things to be sure we optimise correctly. And it remains easy to modify and doesn't impact the rest, so I don't see it as urgent.

Reply:
Finally I kept your proposition, I'm not sure there's a clear gain in propagating the position of the string.

Reply:
Well, it helps to give a more precise position. But it is not very important indeed.

Reply:
At least, you could use mem_sorted instead of just mem.

Reply:
Ah I used the wrong review thing, it's obviously about the use of `cmd.pos` in handle...